liquibase-standard/src/test/groovy/liquibase/changelog/DatabaseChangeLogTest.groovy

```diff
@@ -558,13 +557,70 @@ http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbch
 
         rootChangeLog.load(new ParsedNode(null, "databaseChangeLog")
                 .addChildren([changeSet: [id: "1", author: "nvoxland", createTable: [tableName: "test_table", schemaName: "test_schema"]]])
-                .addChildren([property: [file: "file.properties"]]),
+                .addChildren([property: [file: "file.properties", errorIfMissing: "false"]]),
                 propertiesResourceAccessor)
 
         then:
         rootChangeLog.getChangeLogParameters().hasValue("context", rootChangeLog) == false
     }
 
+    @Unroll
+    def "an error is thrown when properties file is not found and is set to error"() {
+        when:
+        def propertiesResourceAccessor = new MockResourceAccessor(["com/example/file.properties": testProperties])
+
+        def rootChangeLog = new DatabaseChangeLog("com/example/root.xml")
+        rootChangeLog.setChangeLogParameters(new ChangeLogParameters())
+
+        rootChangeLog.load(new ParsedNode(null, "databaseChangeLog")
+                .addChildren([changeSet: [id: "1", author: "nvoxland", createTable: [tableName: "test_table", schemaName: "test_schema"]]])
+                .addChildren([property: [errorIfMissing: errorIfMissingDef, relativeToChangelogFile: relativeToChangelogFileDef, file: fileDef]]),
+                propertiesResourceAccessor)
+
+        then:
+        def e = thrown(UnexpectedLiquibaseException)
+        assert e.getMessage() == FileUtil.getFileNotFoundMessage(fileDef)
+
+        where:
+        errorIfMissingDef    | relativeToChangelogFileDef    | fileDef
+        null                        | null                          | "file.properties"
+        null                        | false                         | "file.properties"
+        null                        | true                          | "com/example/file.properties"
+        true                        | null                          | "file.properties"
+        true                        | false                         | "file.properties"
+        true                        | true                          | "com/example/file.properties"
+    }
+
+    @Unroll
+    def "no error is thrown when properties file is not found and is set to not error and property is not set"() {
```

Comment:
yeah, it might sound a bit confusing and we can update it. I understood exactly what you described in your comment. I can rename it to "no error is thrown when properties file is not found and errorIfMissing" is either set as false or null. @jasonlyle88 do you agree? I'll update as I mentioned, but please feel free to make any other update on top of mine if you have a better name.

Thanks,
Daniel.

Reply:
That is much clearer, thanks for the update. Amazing what makes sense when you are in the middle of something... but when you go back and read it you just question how you ever thought that was a good explanation 🤣 Thanks for the adjustment there!